aws-analytics-pinpoint/src/androidTest/java/com/amplifyframework/analytics/pinpoint/PinpointAnalyticsInstrumentationTest.java

```diff
@@ -0,0 +1,292 @@
+            HubAccumulator.create(HubChannel.ANALYTICS, AnalyticsChannelEventName.FLUSH_EVENTS, 1)
+                .start();
+        AnalyticsEvent event2 = AnalyticsEvent.builder()
+            .name("Amplify-event" + UUID.randomUUID().toString())
+            .addProperty("DemoProperty1", "DemoValue1")
+            .addProperty("DemoProperty2", 2.0)
+            .build();
+
+        Amplify.Analytics.recordEvent(event2);
+        //Autosubmitter runs every 10 secs, so waiting for 15sec to receive the hub event
+        List<HubEvent<?>> hubEvents2 = analyticsHubEventAccumulator1.await(EVENT_FLUSH_TIMEOUT_WAIT, TimeUnit.SECONDS);
+        assertEquals(1, hubEvents2.size());
+    }
+
+    /**
+     * Registers a global property and ensures that all recorded events have the global property.
+     *
+     * @throws JSONException Caused by unexpected event structure.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void registerGlobalPropertiesAddsGivenPropertiesToRecordedEvents() throws JSONException {
+        HubAccumulator analyticsHubEventAccumulator =
+            HubAccumulator.create(HubChannel.ANALYTICS, AnalyticsChannelEventName.FLUSH_EVENTS, 1)
+                .start();
+        // Arrange: Register global properties and create an event
+        Amplify.Analytics.registerGlobalProperties(
+            AnalyticsProperties.builder()
+                .add("GlobalProperty", "GlobalValue")
+                .build()
+        );
+        AnalyticsEvent event = AnalyticsEvent.builder()
+            .name("Amplify-event" + UUID.randomUUID().toString())
+            .addProperty("LocalProperty", "LocalValue")
+            .build();
+
+        // Act: Record two events: the one created above and another just with a key
+        Amplify.Analytics.recordEvent(event);
+        Amplify.Analytics.recordEvent("amplify-test-event");
+        Sleep.milliseconds(RECORD_INSERTION_TIMEOUT);
+        Amplify.Analytics.flushEvents();
+        List<HubEvent<?>> hubEvents = analyticsHubEventAccumulator.await(10, TimeUnit.SECONDS);
+        ArrayList<AnalyticsEvent> submittedEvents = (ArrayList<AnalyticsEvent>) hubEvents.get(0).getData();
+        // Assert: Verify two event were recorded and global attributes are present on both
+        assert submittedEvents != null;
+        assertEquals(2, submittedEvents.size());
+        AnalyticsProperties event1Attributes = submittedEvents.get(0).getProperties();
+        AnalyticsProperties event2Attributes = submittedEvents.get(1).getProperties();
+
+        // Global properties are attached to all events
+        assertEquals("GlobalValue", event1Attributes.get("GlobalProperty").getValue());
+        assertEquals("GlobalValue", event2Attributes.get("GlobalProperty").getValue());
+
+        // Local properties are only attached is passed explicitly as in event
+        assertEquals("LocalValue", event1Attributes.get("LocalProperty").getValue());
```

Comment:
it's weird in the original that there's no equivalent check to make sure that `event2Attributes` doesn't have `LocalProperty`, can we add that here?

Reply:
in followup PR